src/WebJobs.Script/Diagnostics/MetricEvent.cs

```diff
@@ -8,26 +8,45 @@ namespace Microsoft.Azure.WebJobs.Script.Diagnostics
 {
     public abstract class MetricEvent
     {
+        private TimeSpan _duration;
+        private bool _completed;
+
         public string FunctionName { get; set; }
 
         public DateTime Timestamp { get; set; }
 
         public Stopwatch StopWatch { get; set; }
 
-        public TimeSpan Duration { get; set; }
+        /// <summary>
+        /// Gets or sets the total duration for the event.
+        /// </summary>
+        /// <remarks>
+        /// When the event is complete this value should be set to the total duration.
+        /// When set, <see cref="Completed"/> will also return true.
+        /// </remarks>
+        public TimeSpan Duration
+        {
+            get
+            {
+                return _duration;
+            }
+
+            set
+            {
+                _duration = value;
+                _completed = true;
```

Comment:
To minimize changes, I'm maintaining the existing behavior where setting Duration leads to the event being completed. We could also have made things more explicit by requiring Completed to be explicitly set.

Reply:
I really think this would be a lot clearer with a `Complete(TimeSpan duration)` method instead of a settable property. Can we please make that change?